### PR TITLE
Suggested conda go package

### PR DESCRIPTION
**Pull request by sodre**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: .ci_support/linux_.yaml
Comment (sodre) [inline .ci_support/linux_.yaml:10]: This file would be removed once conda-forge/conda-forge-pinning-feedstock#95 is merged.

File: recipe/build.sh
Comment (sodre) [inline recipe/build.sh:13]: All of that is replaced with the multi-source line in the updated meta.yaml file.

File: recipe/build.sh
Comment (sodre) [inline recipe/build.sh]: GOARCH and GOOS are set by the `go_{linux,osx,win}-64` packages if you are using conda's `target_platform`

File: recipe/build.sh
Comment (sodre) [inline recipe/build.sh]: If we agree that GOPATH can always be set to $SRC_DIR, then I can update the activate/deactivate scripts in the compiler specific packages.

File: recipe/meta.yaml
Comment (jakirkham) [inline recipe/meta.yaml]: Probably should bump to `0.1.1` while we are at it.

File: recipe/build.sh
Comment (sodre) [inline recipe/build.sh:23]: License code can be pushed to meta.yaml, thus reducing complexity of build.sh as well.

**Review by sodre — COMMENTED** [review]
I annotated my changes to explain why I think this is a good start for coming up with a default example for go packages.

**Comment by jakirkham** [comment]
Thanks @sodre. Will give this a closer look later.

Do you have any thoughts on how we should handle the `runc` version here?

**Comment by sodre** [comment]
No problem, take your time. I would suggest waiting until conda-forge/conda-forge-pinning-feedstock#95 is merged. 

I can update the runc_version to 0.1.1, but I am not sure what exactly you asking as far as how we should handle it.

**Comment by jakirkham** [comment]
Ok, can wait until that is resolved.

Basically `gosu` is vendoring it's dependencies, which doesn't seem quite right. Didn't know if you knew of a better way to solve this in Go. Alternatively we could include `runc`'s version in this package's version. Not sure if that makes sense either though. General thoughts on handling dependencies with Go would be helpful.

Also you might want to update issue ( https://github.com/conda-forge/go-feedstock/issues/2 ) with the recommendations you have added here. It could wait until the `go` compiler stuff is merged if that would be better.

File: .ci_support/linux_.yaml
Comment (sodre) [reply .ci_support/linux_.yaml:10]: This file is okay here. It was generated by `conda-smithy render.`

File: recipe/build.sh
Comment (sodre) [inline recipe/build.sh:16]: These are replaced by the activate/deactivate scripts in the `go_linux-64` package.

**Comment by sodre** [comment]
Hi @jakirkham, 

I looked at the build code for `gosu`. It is a little awkward that we have to bring `runc` separately, but venturing dependencies is the norm in `go` programs. I believe `gosu` predates tools like `govendor` which make this process simple to maintain. Nowadays, `go` packages such as `kubernetes`, `terraform`, and `etcd` include their vendored dependencies within the source code itself in a `vendor` folder. 

I would not change the version string, but we should make sure that if gosu is ever upgraded to 0.11 that our meta.yaml file matches the `runc` version from their [Dockerfile.](https://github.com/tianon/gosu/blob/master/Dockerfile#L5)

Lastly, there is no point in depending on a future `conda-forge/runc` package since all the go programs I've seen so far are statically compiled.

**Comment by jakirkham** [comment]
Was afraid of that. Well I guess we will embrace the vendoring then.

That's a good suggestion. Will keep that in mind. Thanks.

One advantage might be build times, but AFAIK we are not in any danger of going over with Go packages yet. Though feel free to correct me if I'm wrong.

Does this need to be refreshed now that `go` is a compiler or are we still waiting on some things?

**Comment by sodre** [comment]
I will refresh this recipe and remove the WIP once I'm done.

**Comment by jakirkham** [comment]
Friendly nudge 😉

**Comment by sodre** [comment]
@conda-forge-admin, please rerender

**Comment by sodre** [comment]
@jakirkham, this is ready to merge. Sorry for the delay.

File: recipe/build.sh
Comment (jakirkham) [inline recipe/build.sh:3]: Is this still needed? Meaning I know it is needed for this recipe, but not if it is still needed with the `go` compiler.

File: recipe/build.sh
Comment (jakirkham) [reply recipe/build.sh]: I think this is a sensible default if you agree. Should we raise an issue over at the `go` feedstock to get feedback from others?

File: recipe/build.sh
Comment (jakirkham) [reply recipe/build.sh:23]: Yeah, `conda-build` has gotten a lot smarter since this was written. 😉

File: recipe/meta.yaml
Comment (jakirkham) [reply recipe/meta.yaml]: Let's deal with this independently.

**Comment by jakirkham** [comment]
Thanks @sodre. LGTM.

Had [one comment]( https://github.com/conda-forge/gosu-feedstock/pull/3#discussion_r222023869 ) that we can handle asynchronously and [one non-blocking question above]( https://github.com/conda-forge/gosu-feedstock/pull/3#pullrequestreview-160812820 ). Otherwise I think we should merge.

File: recipe/build.sh
Comment (sodre) [reply recipe/build.sh]: I raised the issue here https://github.com/conda-forge/go-feedstock/issues/31.

File: recipe/build.sh
Comment (sodre) [reply recipe/build.sh:3]: I think this is not needed here. I am removing to see if the code still builds.

Here is the best description I've found about the CGO_ENABLED flag.
https://go-review.googlesource.com/c/go/+/12603/2/src/cmd/cgo/doc.go

Comment (jakirkham) [reply recipe/build.sh:3]: That review makes it sound like we still need to set `CGO_ENABLED=0` if we expect there to be no cgo usage.

FWIW this comes from [here]( https://github.com/tianon/gosu/blob/d2937f478b317e55a77e0e0ed4f79a333a6f5735/Dockerfile#L14-L15 ). I think they are mainly interested in building portable binaries. Though am not clear how they are not still in some way attached to the C runtime library. So this may be less necessary for us.

That said, I think we can table this for now. Was just curious since you know more about Go than I whether this made sense for us. We can save it for another discussion though.

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render.

File: recipe/meta.yaml
Comment (jakirkham) [reply recipe/meta.yaml]: Note to self: This comes from [here]( https://github.com/tianon/gosu/blob/d2937f478b317e55a77e0e0ed4f79a333a6f5735/Dockerfile#L5 ). So we probably shouldn't change it unless upstream does.

**Review by jakirkham — APPROVED** [review]
LGTM. Thanks so much for working on this @sodre. Will merge once CI passes.

**Pull request merged**
